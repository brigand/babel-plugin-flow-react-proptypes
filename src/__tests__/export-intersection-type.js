const babel = require('babel-core');
const content = `
var React = require('react');

type T = {
  bar: string,
}

type U = {
  foo: string,
}

export type V = U & T;


class C extends React.Component {
  props: V;
}

export default C;
`;

it('export-intersection-type', () => {
  const res = babel.transform(content, {
    babelrc: false,
    presets: ['es2015', 'stage-1', 'react'],
    plugins: ['syntax-flow', require('../')],
  }).code;
  expect(res).toMatchSnapshot();
});
