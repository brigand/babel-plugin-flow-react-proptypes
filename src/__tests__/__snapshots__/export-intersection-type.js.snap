// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`export-intersection-type 1`] = `
"'use strict';

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError(\\"Cannot call a class as a function\\"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError(\\"this hasn't been initialised - super() hasn't been called\\"); } return call && (typeof call === \\"object\\" || typeof call === \\"function\\") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== \\"function\\" && superClass !== null) { throw new TypeError(\\"Super expression must either be null or a function, not \\" + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var React = require('react');

var babelPluginFlowReactPropTypes_proptype_V = {
  foo: require('prop-types').string.isRequired,
  bar: require('prop-types').string.isRequired
};
if (typeof exports !== 'undefined') Object.defineProperty(exports, 'babelPluginFlowReactPropTypes_proptype_V', {
  value: babelPluginFlowReactPropTypes_proptype_V,
  configurable: true,
  enumerable: true
});

var C = function (_React$Component) {
  _inherits(C, _React$Component);

  function C() {
    _classCallCheck(this, C);

    return _possibleConstructorReturn(this, (C.__proto__ || Object.getPrototypeOf(C)).apply(this, arguments));
  }

  return C;
}(React.Component);

C.propTypes = {
  foo: require('prop-types').string.isRequired,
  bar: require('prop-types').string.isRequired
};
exports.default = C;"
`;
